extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php

```diff
@@ -297,4 +297,25 @@ private function checkAccess() {
 			throw new \PermissionsError( self::DISCUSSIONS_LOG_ACTION );
 		}
 	}
+
+	private function getLocationFromIP( $ip, array &$ipCache ) {
+		if (array_key_exists($ip, $ipCache)) {
+			return $ipCache[$ip];
+		}
+
+		$details = json_decode(file_get_contents("http://ipinfo.io/{$ip}/json"));
```

Comment:
What if this site goes down? :)
Try looking at GeoIP library, possibly this https://github.com/Wikia/app/blob/master/lib/vendor/Net/GeoIP.php#L661